### PR TITLE
Support skupper annotation of daemonsets and statefulsets

### PR DESCRIPTION
**Pull request by grs**

### CODE REVIEW

**Review by fgiorgetti — APPROVED** [review]
@grs you make our lives easier this way. Thanks for updating the tests as well.
I have submitted a PR to your branch with a minor cosmetic change, just in case
you find it helpful.

**Review by nicob87 — APPROVED** [review]
Good! looks like the annotations are here to stay

File: api/types/types.go [hunk 1]
```diff
@@ -86,6 +86,11 @@ var ControllerEditPolicyRule = []rbacv1.PolicyRule{
 		APIGroups: []string{"apps"},
 		Resources: []string{"deployments", "statefulsets"},
 	},
+	{
```
Comment (ajssmith) [inline api/types/types.go:89]: should the deploy-watch-*.yaml's under the site controller be updated for daemonsets as well?

**Review by ajssmith — APPROVED** [review]
lgtm, nice enhancements

File: api/types/types.go (hunk 1 above)
Comment (grs) [reply api/types/types.go:89]: No, the site controller doesn't watch them, only the service-controller.

**Pull request merged**